doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst

```diff
@@ -0,0 +1,403 @@
+
+.. _zarco_signal_handler_full:
+
+Complete zboss_signal_handler implementation
+********************************************
+
+In its complete implementation, the ``zboss_signal_handler`` allows the application to control a broader set of basic functionalities, including joining, commissioning, and network formation.
+
+There are in fact cases in which the default handler will not be sufficient and needs to be extended.
+For example, when the application wants to use the procedure of the initiator of finding & binding or use the production configuration feature.
+
+Extending zboss_signal_handler
+================================
+
+If you want to extend ``zboss_signal_handler`` to cover additional functionalities, write signal handler implementation for the required ZBOSS signal.
+For example for a device that initiates a F&B procedure, extend ``zboss_signal_handler`` with a case for ``ZB_BDB_SIGNAL_FINDING_AND_BINDING_INITIATOR_FINISHED``:
+
+.. code-block:: c
+
+    void zboss_signal_handler(zb_bufid_t bufid)
+    {
+        zb_zdo_app_signal_hdr_t   *sg_p  = NULL;
+        zb_zdo_app_signal_type_t  sig    = zb_get_app_signal(bufid, &sg_p);
+        zb_ret_t                  status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+        zb_nwk_device_type_t      role   = zb_get_network_role();
+        zb_bool_t                 comm_status;
+
+        switch (sig) {
+            case ZB_BDB_SIGNAL_FINDING_AND_BINDING_INITIATOR_FINISHED:
+                /* Check status of signal. */
+                if (status == RET_OK) {
+                    /* This signal is received with additional data. Read additional information to get status of F&B procedure. */
+                    zb_zdo_signal_fb_initiator_finished_params_t *fnb_params =
+                            ZB_ZDO_SIGNAL_GET_PARAMS(sg_p, zb_zdo_signal_fb_initiator_finished_params_t);
+
+                    switch (fnb_params->status) {
+                    case ZB_ZDO_FB_INITIATOR_STATUS_SUCCESS:
+                        /* F&B with a Target on the Initiator side is completed with a success. */
+                        break;
+
+                    case ZB_ZDO_FB_INITIATOR_STATUS_CANCEL:
+                        /* F&B on the Initiator side is canceled. */
+                        break;
+
+                    case ZB_ZDO_FB_INITIATOR_STATUS_ALARM:
+                        /* F&B on the Initiator side is finished by timeout. */
+                        break;
+
+                    case ZB_ZDO_FB_INITIATOR_STATUS_ERROR:
+                        /* F&B on the Initiator side finished with a failure. */
+                        break;
+
+                    default:
+                        /* Unrecognised status of F&B procedure. */
+                        break;
+                    }
+                } else {
+                    /* F&B procedure failed. */
+                }
+                break;
```

Comment:
Incomplete code snippet:
 - The switch-case structure does not have the closing bracket.
 - There is no call to the default signal handler.
 - The buffer is not freed.

The alternative to adding `default_signal_handler` call is to write in the text above, that this snippet should be placed above the minimal implementation.